### PR TITLE
add quickstart-react-ts template

### PR DESCRIPTION
**Pull request by ofirge**
typescript compilation works

development is done using 'ts-node-dev' which will reload the server on changes like nodemon and also recompile typescript into js.

### CODE REVIEW

File: apps/quickstart-react-ts/src/middlewares/authentication.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import jwt from 'jsonwebtoken';
+const SHARED_SECRET = process.env.MONDAY_SIGNING_SECRET;
+
+export default async function authenticationMiddleware(req, res, next) {
+  try {
+    let { authorization } = req.headers;
+    if (!authorization && req.query) {
+      authorization = req.query.token;
+    }
```
Comment (DorShakedMonday) [inline apps/quickstart-react-ts/src/middlewares/authentication.ts:9]:
```suggestion
    const authorization = req.headers.authorization || req?.query?.token;
```

File: apps/quickstart-react-ts/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
+  "include": ["./src/app.ts"],
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+    "incremental": true,
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
```
Comment (DorShakedMonday) [inline apps/quickstart-react-ts/tsconfig.json:7]: remove

**Review by DorShakedMonday — CHANGES_REQUESTED** [review]

File: apps/quickstart-react-ts/package.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "monday-integration-quickstart-app",
+  "version": "0.0.1",
+  "license": "UNLICENSED",
+  "scripts": {
+    "start": "yarn stop && concurrently \"yarn server\" \"yarn expose\"",
+    "server": "ts-node-dev ./src/app.ts",
+    "TODO/server-with-debugger": "use ts-node-dev",
+    "server-with-debugger": "nodemon --delay 100ms --exec 'fuser -k 9228/tcp; node --inspect=0.0.0.0:9228 ./src/app.js'",
```
[the remark this answers is not in the document]
Comment (ofirge) [reply apps/quickstart-react-ts/package.json:9]: added a script for prod that used ts-node and removed the todo

File: apps/quickstart-react-ts/package.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    "expose": "ngrok http 8302",
+    "stop": "kill-port 8302 && kill-port 4049 && kill-port 4040"
+  },
+  "nodemonConfig": {
+    "watch": [
+      ".env",
+      "src/"
+    ]
+  },
```
[the remark this answers is not in the document]
Comment (ofirge) [reply apps/quickstart-react-ts/package.json:18]: removed

File: apps/quickstart-react-ts/package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  "dependencies": {
+    "body-parser": "^1.19.0",
+    "dotenv": "^5.0.1",
+    "express": "^4.17.1",
+    "jsonwebtoken": "^8.5.1",
+    "monday-sdk-js": "0.0.12",
+    "mysql2": "^2.1.0",
+    "proxyquire": "^2.0.1",
+    "querystring": "^0.2.0",
+    "sequelize": "^5.21.5"
```
[the remark this answers is not in the document]
Comment (ofirge) [reply apps/quickstart-react-ts/package.json:28]: removed

File: apps/quickstart-react-ts/src/middlewares/authentication.ts (hunk 1 above)
Comment (ofirge) [reply apps/quickstart-react-ts/src/middlewares/authentication.ts:9]: did something in the same spirit, also made changes to return more accurate errors

File: apps/quickstart-react-ts/tsconfig.json (hunk 2 above)
Comment (ofirge) [reply apps/quickstart-react-ts/tsconfig.json:7]: kk

**Comment by yuhgto** [comment]
This PR is stale – closing it. Let me know if you want to reopen or these changes are still relevant

**Pull request closed without merging**